### PR TITLE
[Issue #44] finished implementing login on client side

### PR DESCRIPTION
**Pull request by tosinoni**

### CODE REVIEW

File: SECP-service/src/main/resources/assets/app/index.html [hunk 1]
```diff
@@ -32,9 +32,13 @@
                 <!-- Collect the nav links, forms, and other content for toggling -->
                 <div class="collapse navbar-collapse navbar-ex1-collapse">
                     <ul class="nav navbar-nav">
-                        <li>
+                        <li ng-cloak ng-show="!isAuthenticated">
```
Comment (mohameddahrouj) [inline SECP-service/src/main/resources/assets/app/index.html]: this is clean

File: SECP-service/src/main/resources/assets/app/js/app.js [hunk 2]
```diff
@@ -21,6 +36,7 @@ angular.module('SECP', ['ngResource', 'ngRoute', 'ui.bootstrap', 'ui.date', 'rou
         templateUrl: 'views/chat/chats.html',
         controller: 'ChatController',
         css: 'css/chat.css',
+        requiresLogin: true
```
Comment (mohameddahrouj) [inline SECP-service/src/main/resources/assets/app/js/app.js]: This is neat. Is this attribute built into angular-routes?

File: SECP-service/src/main/resources/assets/app/js/login/login-controller.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
 angular.module('SECP')
-    .controller('LoginController', ['$scope', function ($scope) {
-    }]);
\ No newline at end of file
+    .controller('LoginController', ['$scope', 'Auth', '$location', function ($scope, Auth, $location) {
+        $scope.login = function () {
+            Auth.login($scope.user).then(function(res) {
+                if(res.status == 200) {
+                    swal('Welcome to SECP!', 'Login successful!','success');
```
Comment (mohameddahrouj) [inline SECP-service/src/main/resources/assets/app/js/login/login-controller.js]: ah yeeeeah, swal!

**Comment by mohameddahrouj** [comment]
Just thinking out loud but for the admin portal...later on....we can add a very simple check to validate if the user is an admin along with being authenticated and return the views based on that logic

File: SECP-service/src/main/resources/assets/app/js/app.js (hunk 2 above)
Comment (tosinoni) [reply SECP-service/src/main/resources/assets/app/js/app.js]: No this attribute is built into angular jwt. :)

**Comment by tosinoni** [comment]
Yes for the admin portal. You’ll need two things.

1. You need to add requireLogin for the portal route in the app.js.
2. When a user logs in, he is assigned a login role. It could be either normal or admin role. You just need to check the localstorage for the login role property. Then you can determine if the user is an admin or not. :)

File: SECP-service/src/main/resources/assets/app/index.html [hunk 4]
```diff
@@ -32,9 +32,13 @@
...
                             <a href="/login">Login</a>
                         </li>
+
+                        <li ng-cloak ng-show="isAuthenticated" ng-click="logout()">
+                            <a href="/">Logout</a>
+                        </li>
                     </ul>
```
Comment (mohameddahrouj) [inline SECP-service/src/main/resources/assets/app/index.html]: @lavatahir is working on the profile page, when he is done we can add a profile icon to link to a users profile page when authenicated in the navbar

File: SECP-service/src/main/resources/assets/app/index.html [hunk 5]
```diff
@@ -32,9 +32,13 @@
...
                 <div class="collapse navbar-collapse navbar-ex1-collapse">
                     <ul class="nav navbar-nav">
-                        <li>
+                        <li ng-cloak ng-show="!isAuthenticated">
                             <a href="/login">Login</a>
                         </li>
+
+                        <li ng-cloak ng-show="isAuthenticated" ng-click="logout()">
+                            <a href="/">Logout</a>
+                        </li>
```
Comment (mohameddahrouj) [inline SECP-service/src/main/resources/assets/app/index.html]: @lavatahir is working on the profile page, when he is done we can add a profile icon to link to a users profile page when authenicated in the navbar

Comment (tosinoni) [reply SECP-service/src/main/resources/assets/app/index.html]: Good call :)

**Pull request merged**